File: client-android/lib/src/main/kotlin/stasis/client_android/lib/model/core/CrateStorageReservation.kt
package stasis.client_android.lib.model.core

import com.squareup.moshi.JsonClass
import java.util.UUID

@JsonClass(generateAdapter = true)
data class CrateStorageReservation(
    val id: CrateStorageReservationId,
    val crate: CrateId,
    val size: Long,
    val copies: Int,
    val origin: NodeId,
    val target: NodeId
)

typealias CrateStorageReservationId = UUID

File: client-ios/lib/Sources/StasisClientLib/Model/Core/CrateStorageReservation.swift
import Foundation

public typealias CrateStorageReservationId = UUID

public struct CrateStorageReservation: Sendable, Equatable, Hashable, Codable {
    public let id: CrateStorageReservationId
    public let crate: CrateId
    public let size: Int64
    public let copies: Int
    public let origin: NodeId
    public let target: NodeId

    public init(
        id: CrateStorageReservationId,
        crate: CrateId,
        size: Int64,
        copies: Int,
        origin: NodeId,
        target: NodeId
    ) {
        self.id = id
        self.crate = crate
        self.size = size
        self.copies = copies
        self.origin = origin
        self.target = target
    }
}
